### PR TITLE
fla/gdn: scratch-free triple-fused KKT kernel

### PR DESCRIPTION
**Pull request by roberteg16**
## Summary

The triple-fused `(kkt ∘ solve_tril ∘ recompute_w_u)` GDN kernel (`vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py`) currently builds the full `[64, 64]` `(β·k)·kᵀ` product, masks away the upper triangle, writes the result to a per-program fp32 main-memory scratch, and re-reads ten `[16, 16]` sub-blocks from that scratch for the Gauss-Jordan inversion. The scratch round-trip costs ~3.85 MB of main-memory traffic per call at the trace shape, requires a per-call allocation, and is the reason ~37.5% of the KKT-phase MFMAs are spent computing upper-triangular blocks that are immediately discarded. It also opens an aux-stream allocator aliasing window when the MoE shared-experts path freshly-frees memory on an auxiliary stream that PyTorch's caching allocator then hands back to this kernel's scratch.

This PR removes the scratch entirely:

- KKT phase builds the 10 strict-lower `[16, 16]` accumulators directly, loading k per 16-row band so each block is the dot of two bands. Reduction order matches the unfused chain per block, so values are bit-equivalent to the corresponding sub-tiles of the full product.
- Per-block g-diff scaling and masks replace the single full-tile scale+mask. Diagonal blocks apply the in-block strict-lower triangle mask plus row/col boundary; off-diagonal blocks need only the row/col boundary mask (rows are strictly below cols by band construction).
- Gauss-Jordan inversion sources row `i` of each diagonal block in register via `tl.sum(tl.where((o_i == i)[:, None], b_A_kk, 0), 0)`, which produces the same 16 fp32 values an explicit row load would. Algorithm and rounding unchanged.
- `A_scratch` is dropped from the kernel signature and the Python wrapper. The aux-stream aliasing hazard is closed at the source (no scratch buffer to alias) rather than masked by a per-call zero-init.

## What this buys

- ~37.5% fewer MFMAs in the KKT phase (6 upper-triangular blocks of the `[64, 64]` product are no longer computed).
- ~3.85 MB of main-memory scratch traffic per call eliminated at the trace shape, plus the per-call allocation overhead.
- One fewer per-program scratch buffer to size, allocate, or guard against cross-stream allocator races.

## Test plan

- [x] Unit correctness vs the unfused chain (`chunk_scaled_dot_kkt_fwd` → `solve_tril` → `recompute_w_u_fwd`):

  ```
  pytest tests/kernels/test_fused_wy_chain.py -v
  # tests/kernels/test_fused_wy_chain.py::test_triple_fused_wy_matches_unfused PASSED
  # atol=5e-4 / rtol=1e-3 vs unfused chain
  ```

- [x] End-to-end multimodal sanity on Qwen3.5-35B-A3B-W4A16 (the configuration that originally surfaced the aux-stream aliasing hazard), with `VLLM_FLA_USE_FUSED_KKT=1` and shared-experts multi-stream overlap enabled. Engine starts cleanly, generated tokens are coherent, no `hipErrorIllegalAddress`. Bench numbers on ROCm gfx942:

  | Metric             | Before        | After         |
  | ------------------ | ------------- | ------------- |
  | TTFT (ms)          | 337.68        | 335.67        |
  | TPOT (ms)          | 12.35         | 12.32         |
  | Output tok/s       | 67.16         | 67.33         |

  (Numbers within noise; the win is correctness/cleanliness, not throughput.)

### CODE REVIEW

**Review by mgehre-amd — APPROVED** [review]

File: vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py [hunk 1]
```diff
@@ -458,7 +458,16 @@ def fused_kkt_solve_tril_recompute_w_u_fwd(
...
+    # NB: torch.zeros (not torch.empty) — under cudagraph capture with the MoE
+    # shared-experts running on an aux stream, torch.empty's caching allocator
+    # was handing out memory blocks that aux-stream tensors had just freed,
+    # leaving stale cache lines that the kernel's intra-program write→read
+    # round-trip would occasionally re-read instead of the freshly-stored b_A.
+    # zero-init forces the allocator to clean the block and breaks the
+    # aliasing window.  Cost: a single bf16-bandwidth fill per call (~3.85 MB
+    # / 230 GiB/s ≈ 17 µs at trace shape) — small vs the prefill speedup
+    # this fusion unlocks.
+    A_scratch = torch.zeros(B, T, H, BT, dtype=torch.float32, device=k.device)
```
Comment (mgehre-amd) [inline vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py]: To investigate more deeply whether this is a runtime/hip graph or other issue.

File: vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py [hunk 2]
```diff
@@ -4,12 +4,18 @@
...
+also saves the per-program HBM scratch (~3.85 MB at the trace shape):
+the 10 strict-lower 16×16 blocks of (β·k)·kᵀ are accumulated directly
+in registers, skipping both the [64,64]→scratch→10×[16,16] dance and
+the 6 upper-triangular MFMAs that would otherwise be computed only to
+be masked away (≈37.5% FLOP reduction in the KKT phase).
 
 Inputs:
     k, v, beta, g_cumsum   — same as the unfused chain
     NO A tensor — computed in-kernel from k * β * kᵀ
+    NO HBM scratch — sub-blocks live in registers end-to-end
```
Comment (mgehre-amd) [inline vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py]: This comment is not useful

File: vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py [hunk 3]
```diff
@@ -4,12 +4,18 @@
 """Triple-fused (kkt ∘ solve_tril ∘ recompute_w_u) kernel.
 
 Builds on wy_fast_fused.py by additionally absorbing the upstream
-``chunk_scaled_dot_kkt_fwd`` kernel.  Saves the A writeback (~3.85 MB
-at the trace shape), the A readback, plus one launch.
+``chunk_scaled_dot_kkt_fwd`` kernel.  Saves the A writeback, the A
+readback, and one launch.  Compared with the prior round-trip variant,
```
Comment (mgehre-amd) [inline vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py]: "Compared with the prior round-trip variant,": This comment is not useful

File: vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py (hunk 2 above)
Comment (roberteg16) [reply vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py]: Gone

File: vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py (hunk 3 above)
Comment (roberteg16) [reply vllm/model_executor/layers/fla/ops/wy_fast_doubly_fused.py]: Gone
